### ISSUE TITLE
Secondary Subnet and Routability

### ISSUE THREAD

**Issue by rhysxevans** [aws/amazon-vpc-cni-k8s#1359]
Hi

I may be missing something , but here I go, so any help is appreciated.

So my setup is as follows

VPC(me) -> TGW -> VPC -> NGW -> IGW

VPC(me) has 2 subnets per AZ, Routable for Instances and Non route-able for PODs

The network only knows about the 192.168.0.0/16 range and we only have a /24 out of that
We are not allowed to update routes on this network and are not allowed to provision dedicated egress points 

I have spun EKS up (1.18 eks.3) and configured a secondary subnet (100.64.0.0/10) as per https://aws.amazon.com/premiumsupport/knowledge-center/eks-multiple-cidr-ranges/ , EXCLUDING the routing bit on the new subnets, and I get pods coming up on this range. However I am unable to get pod traffic destined for outside (NOT 192.168.0.0/16 or 100.64.0.0/10) to to flow.

With this CNI is there any way to accomplish this ?  

Basically if traffic is not destined for the POD network/subnet, route via the instances primary ENI, in the primary routeable subnet, and masq/nat the traffic to the primary ENI's primary IP 

Thanks for the response in advance

**Comment by jayanthvn** [comment]
Hi @rhysxevans 

Can you please share the below information - 

1. Instance id to verify route table entries
2. ENI config
3. aws-node ds o/p

You can either share here or email me at varavaj@amazon.com

Thank you!

**Comment by jayanthvn** [comment]
Also can you please confirm if `AWS_VPC_K8S_CNI_EXTERNALSNAT` is set to false?

**Comment by rhysxevans** [comment]
Hi

I have not changed that value,

      ADDITIONAL_ENI_TAGS:                 {}
      AWS_VPC_CNI_NODE_PORT_SUPPORT:       true
      AWS_VPC_ENI_MTU:                     9001
      AWS_VPC_K8S_CNI_CONFIGURE_RPFILTER:  false
      AWS_VPC_K8S_CNI_CUSTOM_NETWORK_CFG:  true
      AWS_VPC_K8S_CNI_EXTERNALSNAT:        false
      AWS_VPC_K8S_CNI_LOGLEVEL:            DEBUG
      AWS_VPC_K8S_CNI_LOG_FILE:
 /host/var/log/aws-routed-eni/ipamd.log
      AWS_VPC_K8S_CNI_RANDOMIZESNAT:       prng
      AWS_VPC_K8S_CNI_VETHPREFIX:          eni
      AWS_VPC_K8S_PLUGIN_LOG_FILE:
/var/log/aws-routed-eni/plugin.log
      AWS_VPC_K8S_PLUGIN_LOG_LEVEL:        DEBUG
      DISABLE_INTROSPECTION:               false
      DISABLE_METRICS:                     false
      ENABLE_POD_ENI:                      false
      MY_NODE_NAME:                         (v1:spec.nodeName)
      WARM_ENI_TARGET:                     1
      ENI_CONFIG_LABEL_DEF:
failure-domain.beta.kubernetes.io/zone

Thanks

**Comment by jayanthvn** [comment]
Hi @rhysxevans 

Can you please share your clusterARN? You can email it to me - varavaj@amazon.com

**Comment by jayanthvn** [comment]
Hi @rhysxevans 

Can you please set SNAT to true and confirm if that fixes the issue.

**Comment by rhysxevans** [comment]
Hi

Closing this, we have moved to Calico as it covers this use case and our pod density requirements

Thanks
